### PR TITLE
GraphQL User queries and mutations

### LINKED ISSUE #15: User GraphQL

### PR DESCRIPTION
**Pull request by nicorithner**
### Participating Group Members:

*Saryn Mooney
* Daniel Lessenden
* Nico Rithner
 
### Code Highlights:

   * Created graphql user type
   * Created graphql user queries
   * Created graphql  user mutations
 
### Have Tests Been Added?

- [ ] No.
- [ ] Yes, but all tests are not passing. 
- [x] Yes, and all are passing.
 
### Any background context you want to provide?
 
In the commits you'll notice that I took an approach and then backed away from it. So there is some file creation and deletion there.
To get the tests going I did one with Tyler and that was helpful. He helped me understand how to structure the requests.
There is test graphql queries that is not a request. I did it and Tyler said it was fine but not necessary. 

### Issues: 
* Addresses #
* Closes #
 
 
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] All new and existing tests passed  - [ ] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,8 @@ gem 'graphql', '1.9.18'
 group :development, :test do
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   gem 'capybara'
+  gem 'factory_bot_rails'
+  gem 'faker'
   gem 'pry'
```
Comment (D-Lessenden) [inline Gemfile:21]: Great idea adding faker.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+  validates :username, :email, uniqueness: true, presence: true
+  validates :phone, presence: true
   validates :avatar, presence: true
   has_many :group_users
   has_many :groups, through: :group_users
   has_many :checkins
 
-  enum role: ['caretaker', 'lone_ranger']
+  enum role: ['user', 'caretaker', 'lone_ranger']
 end
```
Comment (D-Lessenden) [inline app/models/user.rb:10]: What is the advantage to add "user" as a role. Is it just to have a neutral default setting?

**Review by D-Lessenden — APPROVED** [review]
Good work on this, I approved it but will leave it unmerged so that Saryn can take a look as well.

**Review by sarynm12 — APPROVED** [review]
This looks good, Nico! Thank you!

**Pull request merged**